data-model-layer/data-model.md

```diff
@@ -115,6 +115,8 @@ Map is a recursive kind.
 Values in maps are accessed by their "key".  Maps can also be iterated over,
 yielding key+value pairs.
 
+Maps must be serialized in a deterministic way (key sorting should be consistent regardless of insertion order or language specific sorting). The sorting is not defined by the data model, each codec should find the best sorting algorithm, but the sorting must be consitent between languages and is most often implemented as additional code during serialization since most formats do not already have deterministic maps.
```

Comment:
I would actually phrase this very differently.  We're in the Data Model specs.  The word "serialization" *shouldn't appear* except in saying something is a Codec concern.

```suggestion
Map iteration order should be stable.
This is important because we expect that when using a Codec to deserialize data,
handle it in the standard ways the Data Model supports,
and then serialize it again using the Codec,
it should be possible to serialize it again with a minimal change to the serial form;
defining map iteration order as stable in the Data Model gives us this ability.

(Note that many Codecs define canonical sorting for map serialization -- but
this is a concern of those Codecs, not of the Data Model.)
```

Reply:
I find @mikeal's version clearer. Here's my suggestion to combine both:

```suggestion
Codecs must be able to serialize Maps in a deterministic way (key sorting should be consistent regardless of insertion order or language specific sorting). The sorting is not defined by the data model, each codec should find the best sorting algorithm, but the sorting must be consistent between languages and is most often implemented as additional code during serialization since most formats do not already have deterministic maps.
```